### ISSUE TITLE
What is the state of this project? 

### ISSUE THREAD

**Issue by DeflateAwning** [secworks/blake2#4]
I would like to incorporate this project into another project. I'm having trouble telling how close to complete and functional this project is, though. Can you summarize what blocks still need to be designed/implemented, and I will do my best to work on them. Has development on this project stopped entirely? Thanks!

**Comment by secworks** [comment]
Aloha!

As the status says: "Not done. Does Not Work. Do. Not. Use"

To be less flippant (sorry), it really is quite far from bering completed. Somebody did take a stab and pushed fixes. But that core did not generate correct results and did not build properly. And the code was quite far from the style and quality I try to use.

The development has been paused for quite some time. But as it happened I actually looked at this project (and the blake2s project) this weekend. With the conclusion of the poly1305 core, completing the blake cores is probably next on my list for 2020. But my schedule is unpredictable. I *may* do them this year.

As it is right now, I don't know the state. Taking a fast glance at the testbenches, the top level wrapper has not been worked on. The core has some weird init-part that seems like a test too. Running the testcases indicate that the G function is probably done. But m-select is not done yet. Just by this fast check I would probably need to spend an afternoon to reaquaint myself and get a feeling for the state. So Not Done is probably a correct assessment ;-)

I would say that all blocks needs to be worked on. the top level wrapper (being pretty much a std interface) is probably the one in best shape. But if you are going to interface to AXI for example, you probably would not use that file anyeway.


The plan was to complete blake2s first.
Is it blake2b or blake2s you are planning to use?

**Comment by secworks** [comment]
Just to elaborate. If the testbenches aren't working and contains real testcases, the corresponding RTL source is probably not completed and ready to be used.

**Comment by DeflateAwning** [comment]
Turns out I opened this issue on the wrong repo. I meant to open it on the Blake2s one, but it looks like you answered for both projects (thanks).

Sounds neat, and am looking forward to seeing more progress, if you get around to it. I might play around with the Blake2s a bit and see what I can accomplish! Thanks again

**Comment by secworks** [comment]
I realized that I had two blake2s repos. One basically empty public repo. And one private with a lot more work done. I've now moved the private repo into the public. It is still not done, but there is much more real stuff in the repo now.

https://github.com/secworks/blake2s

One thing I've done is to add a local copy of the reference C implementation. This allows me to add a lot of printfs to get intermediate results etc.

If you have issues, ideas we cam continue the discussion with that repo.
